activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1731,6 +1731,8 @@ def check_if_method_has_arguments!(method_name, args, message = nil)
         [:extending, :where, :having, :unscope, :references, :annotate, :optimizer_hints]
       ).freeze # :nodoc:
 
+      STRUCTURAL_VALUE_METHODS_TO_IGNORE_ORDER = [:eager_load, :includes, :joins, :left_outer_joins, :preload, :select].freeze # :nodoc:
```

Comment:
This name doesn't make sense to me. It says "to ignore order" but then we sort below, essentially the opposite of ignoring order because we set an explicit order. I think this needs a better (and shorter) name.

Also in your issue it said "This problem exists for all methods provided in STRUCTURAL_VALUE_METHODS constant in ActiveRecord::QueryMethods. I think for most of them arguments order should not be checked." but this isn't most of the methods in STRUCTURAL_VALUE_METHODS, so are we missing some? or was that assertion incorrect?

Reply:
Thank you for your review @eileencodes 

> This name doesn't make sense to me. It says "to ignore order" but then we sort below, essentially the opposite of ignoring order because we set an explicit order. I think this needs a better (and shorter) name.

Sure, I'll think on better name, yeah my bad on incorrect logic to verify the order. 

> Also in your issue it said "This problem exists for all methods provided in STRUCTURAL_VALUE_METHODS constant in ActiveRecord::QueryMethods. I think for most of them arguments order should not be checked." but this isn't most of the methods in STRUCTURAL_VALUE_METHODS, so are we missing some? or was that assertion incorrect?

No when I am reading your comment I feel we should ignore the order on those, I think I missed on those due to incorrect assumption. I'll explore it in this week.